experiments/dorop-20/metadata/element_adapter.py

```diff
@@ -0,0 +1,46 @@
+import xml.etree.ElementTree as ET
+from xml.etree.ElementTree import Element
+
+from metadata.exceptions import DataNotFoundError
+
+class ElementAdapter():
+
+    @classmethod
+    def from_string(cls, text: str, namespaces: dict[str, str]) -> "ElementAdapter":
+        return cls(ET.fromstring(text=text), namespaces)
+
+    def __init__(self, elem: Element, namespaces: dict[str, str]):
+        self.elem = elem
+        self.namespaces = namespaces
+
+    def find(self, path: str) -> "ElementAdapter":
+        result = self.elem.find(path, self.namespaces)
+        if result is None:
+            raise DataNotFoundError(f"No element found for path {path}")
```

Comment:
@ssciolla are these exceptions the reason for `ElementAdapter`?

Reply:
Pretty much. The adapter provides a couple other advantages (stores namespaces and shares them with children, reduces and specifies the API surface of the element-like object we deal with), but the primary one is that I can raise errors if we don't find what we expect. See https://github.com/mlibrary/dor-py/pull/7#discussion_r1847394194

There are probably other ways to solve this type safety problem, but this is the one I chose for now.